python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_categorical.py

```diff
@@ -0,0 +1,375 @@
+        - 0.0870775 * x2
+        - 0.106959 * x3
+        + 7.98772 * x3 ** 3 * 1e-6
+        + 0.00242482 * x4
+        + 1.32851 * x4 ** 3 * 1e-6 * 0.00146393 * x1 * x2
+        - 0.00301588 * x1 * x3
+        - 0.00272291 * x1 * x4
+        + 0.0017004 * x2 * x3
+        + 0.0038428 * x2 * x4
+        - 0.000198969 * x3 * x4
+        + 1.86025 * x1 * x2 * x3 * 1e-5
+        - 1.88719 * x1 * x2 * x4 * 1e-6
+        + 2.50923 * x1 * x3 * x4 * 1e-5
+        - 5.62199 * x2 * x3 * x4 * 1e-5
+    )
+    return y
+
+
+def Goldstein(inp):
+    x1, x2, z1, z2 = inp
+    x1 = 100 * x1
+    x2 = 100 * x2
+
+    if z1 == 0:
+        x3 = 80
+    elif z1 == 1:
+        x3 = 20
+    elif z1 == 2:
+        x3 = 50
+    else:
+        print("error, no matching category z1")
+
+    if z2 == 0:
+        x4 = 20
+    elif z2 == 1:
+        x4 = 80
+    elif z2 == 2:
+        x4 = 50
+    else:
+        print("error, no matching category z2")
+
+    return [h(x1, x2, x3, x4)]
+
+
+dim = 4
+fun = ot.PythonFunction(dim, 1, Goldstein)
+lvls1 = 3  # Number of discrete levels for z1
+lvls2 = 3  # Number of discrete levels for z2
+# Input distribution
+dist = ot.ComposedDistribution(
+    [
+        ot.Uniform(0, 1),
+        ot.Uniform(0, 1),
+        ot.UserDefined(np.arange(0, lvls1).reshape((-1, 1))),
+        ot.UserDefined(np.arange(0, lvls2).reshape((-1, 1))),
+    ]
+)
+
+# %%
+# In this example, we compare the performances of the LatentVariableModel with a naive
+# approach, which would consist in modeling each combination of categorical
+# variables through a separate and independent Gaussian process.
+
+# %%
+# In order to deal with mixed continuous / discrete problems we can rely on the
+# ProductCovarianceModel class. We start here by defining the product kernel,
+# which combines SquaredExponential kernels for the continuous variables, and
+# LatentVariableModel for the discrete ones.
+
+# %%
+latdim = 2  # Dimension of the latent space
+activecoord = (
+    2 + latdim * (lvls1 - 2) + latdim * (lvls2 - 2)
+)  # Nb ative coordinates in the latent space
+kx1 = ot.SquaredExponential(1)
+kx2 = ot.SquaredExponential(1)
+kz1 = otexp.LatentVariableModel(lvls1, latdim)
+kz2 = otexp.LatentVariableModel(lvls2, latdim)
+kLV = ot.ProductCovarianceModel([kx1, kx2, kz1, kz2])
+kLV.setNuggetFactor(1e-6)
+# Bounds for the hyperparameter optimization
+lb_LV = [1e-4] * dim + [-10] * activecoord
+ub_LV = [3.] * dim + [10.] * activecoord
+bounds_LV = ot.Interval(lb_LV, ub_LV)
+# Distribution for the hyperparameters initialization
+initdist_LV = ot.DistributionCollection()
+for i in range(len(lb_LV)):
+    initdist_LV.add(ot.Uniform(lb_LV[i], ub_LV[i]))
+initdist_LV = ot.ComposedDistribution(initdist_LV)
+
+# %%
+# Alternatively, we consider a purely continuous kernel for independent Gaussian processes.
+# One for each combination of categorical variables levels.
+
+# %%
+k_ind = ot.SquaredExponential(2)
+lb_ind = [1e-4, 1e-4]
+ub_ind = [3., 3.]
```

Comment:
Please use [Black](https://pypi.org/project/black/) on your script:
```suggestion
ub_ind = [3.0, 3.0]
```